### DISCUSSION TITLE
Nothing appears on Virtual serial consol

### DISCUSSION THREAD

**Discussion by b4ldr** [PartialVolume/shredos.x86_64#264] [Q&A]
Hello All,

Im trying to set a boot option in our servers that will allow use to automatically wipe a machine when we decommission.  I have been testing ShredOS, it works great so far, however I do have one issue.  Unfortunately, the console redirection to the idrac virtual serial (/dev/ttyS1) is not working.  For people familiar with Dell servers, what I mean by the virtual console is doing the following
```
$ ssh idra.example.org
>>console com2
```
Logging in to the idrac web interface and starting the web console works (which i believe is /dev/ttyS0)

Here is the grub config im using
```
menuentry 'ShredOS' {
  search --no-floppy --fs-uuid --set=root 4ae45437-45d9-464e-a32f-fff2797066dc
  set iso_path=/iso/shredos.iso
  loopback loop "$iso_path"
  root=(loop)
  linux /boot/bzImage boot=live console=ttyS0 console=ttyS1,115200n8 loglevel=9 findiso=$iso_path toram noeject nwipe_options="--nousb --nowait --autonuke --autopoweroff"
}
```
I have also tried using `--nogui` and get the same issue.  It also seems like the Virtual keyboard is not working as im unable to change to tty2.  Is anyone able to offer help or advice thanks

**Comment by PartialVolume** [comment]
Just to clarify. You can view and control the default nwipe that ShredOS boots (on /dev/tty1) via the idrac virtual server, it's just switching virtual terminals doesn't work?

**Reply by b4ldr** [reply]
Thanks for the response.  No, that's not quite it.  On /dev/ttyS1 i get nothing at all, No text and no keyboard.  /dev/ttyS1 is a virtual serial provided by iDRAC (the dell BMC).

The servers also have a second virtual console accessible via the iDRAC web interface.  I believe this uses /dev/ttyS0.  On this second web interface virtual console, im able to get output and a can see that everything is working.  However, the virtual keyboard is not working (i wanted to alt+f2 so i could see if i could see if there was even a /dev/ttyS1 device)

I appreciate this is likely an edge case and unfortunately I'm unable to provide you with a dell server to recreate as such any guidance is greatly appreciated :)

**Reply by PartialVolume** [reply]
What you are trying to do reminds me of a little project I set up where I wanted to send the data in /dev/tty3 to a USB serial port. Basically two computers both running ShredOS and using minicom on the second system could communicate with the first system via USB. I'm not sure my notes will help you much but for what it's worth I've included them below.  

Sounds to me you want to redirect /dev/tty1 which is the device that nwipe is launched in and redirect it to /dev/ttyS0 ? /dev/tty2 is the virtual window running bash as is /dev/tty3 but is also the system console. For reference /dev/tty1 is not running a bash interpreter, i.e if you were to kill the nwipe process you wouldn't get a bash prompt in /dev/tty1.

I'm just wondering whether you need to play with the settings in inittab that decides where the nwipe launcher launches nwipe, maybe changing it from /dev/tty1 to /dev/ttyS0

Take a look at line 17 https://github.com/PartialVolume/shredos.x86_64/blob/master/board/shredos/fsoverlay/etc/inittab

Of course, if your edited this you would need to build a new .img or .iso

If you exhaust other possibilities and didn't want to build from source I could always make the change and send the the .iso to test it.

```
This does work, but obviously only starts sending console messages after the USB devices have been initialised.

Append this to the kernel command line
console=ttyUSB0,115200N8 console=tty3

Connect a crossover serial link between the two systems.

Use minicom, switch on local echo and add line feed.
i.e CNTRL A E and CNTRL A A

I have got the serial connection working between two computers using minicom on both systems.

Tested with  ShredOS  sending the console output via RS232 to another computer running ShredOS.
```

**Reply by PartialVolume** [reply]
Having re-read your post I may have meant /dev/ttyS1 not /dev/ttyS0.

I don't really understand the workings of idrac so hopefully I'm not misunderstanding this. I read it's a controller board installed in the system?

**Reply by b4ldr** [reply]
@PartialVolume thanks for the response, The instructions you have provided sounds about right to me.  I had hoped to avoid rebuilding the iso but that's not too big a deal.  However, as this is not a quick fix and things work mostly good enough, I may not get the chance to test this for a week or so.   Ill report back once i have tested but feel free to close this down for now if that makes your work flow easier.

Thanks for the speedy support :)

**Comment by Firminator** [comment]
>I believe this uses /dev/ttyS0. On this second web interface virtual console, im able to get output and a can see that everything is working. However, the virtual keyboard is not working (i wanted to alt+f2 so i could see if i could see if there was even a /dev/ttyS1 device)

Random thought:  I recall that there is a setting somewhere in the BIOS or the iDRAC that allows you to configure Serial Redirect. Might want to play with these settings. Sounds like a similar issue mentioned in issue https://github.com/PartialVolume/shredos.x86_64/issues/294
